tests/test_sansio.py

```diff
@@ -521,14 +521,20 @@ def test_no_ratelimit(self):
 
 
 class TestFormatUrl:
-    def test_absolute_url(self):
-        original_url = "https://api.github.com/notifications"
-        url = sansio.format_url(original_url, {})
-        assert url == original_url
-
-    def test_relative_url(self):
-        url = sansio.format_url("/notifications", {})
-        assert url == "https://api.github.com/notifications"
+    @pytest.mark.parametrize(
+        "base_url",
+        ["https://api.github.com/notifications", "https://my.host.com/notifications"],
+    )
+    def test_absolute_url(self, base_url):
```

Comment:
Could we add a test for when the URL is absolute and base_url is also absolute but to a different domain? IOW `https://api.github.com/notifications` with a base URL of `my.host.com/notifications`? I would expect `my.host.com` to win out, but we should probably add a test for that to make sure we don't break that expectation.

Reply:
At the moment if the base url is different than the absolute url, the absolute url wins. 
So if we say 
```
sansio.format_url(
        "https://api.otherhost.com/notifications",
        base_url="https://my.github.com/")
```

it will return `https://api.otherhost.com/notifications`. 

I personally think the passed absolute url should win over the base_url 🤔 
But if we need it the other way around, this will take a bit more work.

Reply:
Don't worry about it then. If it turns out to be a problem then I'm sure someone will tell us. 😉